src/Development/IDE/Core/Compile.hs

```diff
@@ -447,8 +449,51 @@ getModSummaryFromBuffer fp contents dflags parsed = do
     where
       sourceType = if "-boot" `isSuffixOf` takeExtension fp then HsBootFile else HsSrcFile
 
-
--- | Given a buffer, flags, file path and module summary, produce a
+-- | Given a buffer, env and filepath, produce a module summary by parsing only the imports.
+--   Runs preprocessors as needed.
+getModSummaryFromImports
```

Comment:
This looks like it has probably become obsolete by the changes from @pepeiborra that we landed in the meantime.